src/Async/index.js

```diff
@@ -48,14 +48,22 @@ function fromNode(fn, ctx) {
     throw new TypeError('Async.fromNode: CPS function required')
   }
 
-  return (...args) =>
-    Async((reject, resolve) => {
+  const _fn = curry(fn)
+
+  return (...args) => {
+
+    if (args.length < _fn.length - 1) {
+      return fromNode(_fn(...args), ctx)
+    }
```

Comment:
Per your documentation and our discussions their is a chance that the given function has a length of 0. Would this be a case to return an error?

Reply:
My understanding at this point is that if the function has a length of 0 then the safest thing to do is assume that it isn't being partially applied, and apply all of the parameters as per usual. This will mean that any code that currently passes a `compose`d or `...args` or `argument` function will continue to work in exactly the same way as it does today.

Reply:
That's true, sounds good to me